bpf/deserialization_to_bpf_map/deserialization_to_bpf_map.c

```diff
@@ -278,176 +230,194 @@ static int get_map_fd_info(unsigned int id, int *map_fd, struct bpf_map_info *in
     return ret;
 }
 
-static int bpf_get_map_id_by_fd(int map_fd)
+static int free_outter_map_entry(struct op_context *ctx, unsigned int *outer_key)
 {
-    struct bpf_map_info info = {};
-    __u32 info_len = sizeof(info);
+    unsigned char type = MAP_GET_TYPE(*outer_key);
+    unsigned int inner_idx = MAP_GET_INDEX(*outer_key);
 
-    int ret = bpf_obj_get_info_by_fd(map_fd, &info, &info_len);
-    if (ret < 0) {
-        LOG_ERR("bpf_obj_get_info_by_fd failed, map_fd:%d ret:%d ERRNO:%d\n", map_fd, ret, errno);
-        return ret;
-    }
-    return info.id;
+    if (type >= MAP_TYPE_MAX || inner_idx >= MAP_MAX_ENTRIES)
+        return -1;
+
+    CLEAR_BIT(g_map_mng.mim_used_bitmap[type], inner_idx);
+    *outer_key = 0;
+    return 0;
 }
 
-static int free_outter_map_entry(struct op_context *ctx, void *outter_key)
+static int bitmap_find_first_clear(unsigned char *bitmap, int len)
```

Comment:
This is a little too slow, i am thinking if we can use optimistic algorithm, record last allocated bit postion. And at the next time, we allocate from the last position, this can largely reduce the loop and compare

Reply:
Good idea~ It will be optimized.